Classes/Plugin/Eid/SearchInDocument.php

```diff
@@ -38,29 +38,48 @@ class SearchInDocument
      */
     public function main(ServerRequestInterface $request)
     {
+        $output = [
+            'documents' => [],
+            'numFound' => 0
+        ];
+        // Get input parameters and decrypt core name.
         $parameters = $request->getParsedBody();
-        $encrypted = (string)$parameters['encrypted'];
-        $hashed = (string)$parameters['hashed'];
+        $encrypted = (string) $parameters['encrypted'];
+        $hashed = (string) $parameters['hashed'];
+        $count = intval($parameters['start']);
         if (empty($encrypted) || empty($hashed)) {
             throw new \InvalidArgumentException('No valid parameter passed!', 1580585079);
         }
         $core = Helper::decrypt($encrypted, $hashed);
-
-        $output = '';
-        if (!empty($core)) {
-            $query = (string)$parameters['q'];
-            $uid = (string)$parameters['uid'];
-            $start = (string)$parameters['start'];
-            $url = trim(Solr::getSolrUrl($core), '/') . '/select?wt=json&q=fulltext:(' . Solr::escapeQuery($query) . ')%20AND%20uid:' . $uid
-                . '&hl=on&hl.fl=fulltext&fl=uid,id,page&hl.method=fastVector'
-                . '&start=' . $start . '&rows=20';
-            $output = GeneralUtility::getUrl($url);
+        // Perform Solr query.
+        $solr = Solr::getInstance($core);
+        if ($solr->ready) {
```

Comment:
solr instance is not ready at this point.

$solr->config->path is empty. Should be "/".
$solr->config->timeout is 0. Should be 10.
Both is not set in the extension configuration.

Reply:
> $solr->config->path is empty. Should be "/".

This is actually correct. A single `/` is added in `Solr::__construct()` at line 640.

> $solr->config->timeout is 0. Should be 10.

Hm, that was a tricky one. In line 333 of `/Classes/Common/Solr.php` the timeout is set as:
`MathUtility::forceIntegerInRange($conf['solrTimeout'], 1, intval(ini_get('max_execution_time')), 10)`

So, if `$conf['timeout']` is not set in the extension configuration, a default value of `10` is set, but only if this value is between `1` and `max_execution_time`.

The tricky part is, that if you are using something like XDebug or a development environment which has `max_execution_time` disabled, `ini_get('max_execution_time')` will return `0`, thus forcing the timeout to be `0` as well.

I've changed this so that the timeout is limited to 30 seconds if `max_execution_time` is 0.